pallav/fashionTech/pallavBanka/parkingAssignment/Parking.java

```diff
@@ -0,0 +1,122 @@
+package fashionTech.pallavBanka.parkingAssignment;
+
+import java.util.*;
+
+class ParkingCharge {
+    Date inTime;
+    Date outTime;
+    private Long ParkingTime(Date inTime, Date outTime) {
+        this.inTime = inTime;
+        this.outTime = outTime;
+        if (this.outTime != null) {
+            //difference in time in milliseconds
+            Long diff = outTime.getTime() - inTime.getTime();
+            Long diffHours = diff / (60 * 60 * 1000) % 24;
+            return diffHours;
+        } else{
+            System.out.println("The vehicle hasn't left the premises.");
+            return null;
+        }
+    }
+    private Long charges(String s){
+        Hashtable<String, Integer> charges=new Hashtable<String, Integer>();
+        charges.put("TwoWheeler", 20);
+        charges.put("FourWheeler", 40);
+        charges.put("SixWheeler", 80);
+        return charges.get(s)*ParkingTime(inTime, outTime);
+    }
+}
+
+class Floor {
+    private void floor() {
+        Hashtable<Integer, String> vehiclewisefloor = new Hashtable<Integer, String>();
```

Comment:
Use Hashmap

Reply:
Also you should be using Map<String, List<Integer>> for storing vehicleType->List<Floors>
The way you have designed, you will have to traverse the whole map, and check for each floor if parking of 2-Wheeler is allowed or not on that floor.

Reply:
done that